clippy_lints/src/unwrap_in_result.rs

```diff
@@ -0,0 +1,138 @@
+use crate::utils::{
+    is_type_diagnostic_item, method_chain_args, span_lint_and_then, walk_ptrs_ty,return_ty
+};
+use if_chain::if_chain;
+use rustc_hir as hir;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::hir::map::Map;
+use rustc_middle::ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::Span;
+
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for functions of type Result that contain `expect()` or `unwrap()`
+    ///
+    /// **Why is this bad?** Potential errors can be masked.
```

Comment:
```suggestion
    /// **Why is this bad?** These functions promote recoverable errors to non-recoverable errors which may be undesirable in code bases which wish to avoid panics.
```